### PR TITLE
pctl search

### PR DESCRIPTION
**Pull request by aclevername**
Issue: https://github.com/weaveworks/profiles/issues/24

PR based off branch https://github.com/weaveworks/pctl/pull/2

### Description

Adds support for `pctl search`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+In order to test the CLI you need a profiles catalog controller up and running along with its API.
+To get a local setup clone the [Profiles repo](https://github.com/weaveworks/profiles) and run `make local-env`.
+This will deploy a local kind cluster with the catalog controller and API running. Once the environment is setup
+run the following to use pctl against it:
+
+1. Create your catalog, for example there is a `examples/profile-catalog-source.yaml` file in the profiles repo
+`kubectl apply -f profiles/examples/profile-catalog-source.yaml`
+1. In a separate terminal run `kubectl -n profiles-system port-forward <profiles-controller-pod-name> 8000:8000` to enable access to the API
+1. Run `pctl search --catalog-url http://localhost:8000 <profile-name>` to search for your profile
+>>>>>>> 74391c0 (update README)
```
Comment (Callisto13) [inline README.md:25]: womp womp

Comment (aclevername) [reply README.md:25]: :facepalm:

File: cmd/pctl/main.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/urfave/cli/v2"
+	"github.com/weaveworks/pctl/pkg/catalog"
+)
+
+func main() {
+	app := &cli.App{
+		Commands: []*cli.Command{
+			{
+				Name:    "search",
+				Aliases: []string{"a"},
```
[the remark this answers is not in the document]
Comment (aclevername) [reply cmd/pctl/main.go:17]: bad copy pasta, thanks!

File: pkg/catalog/profile_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package catalog_test
```
[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/catalog/profile_suite_test.go:1]: renamings! good spot

File: tests/integration/integration_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		fmt.Fprintf(w, `[
+	{
+		"name": "weaveworks-nginx",
+		"description": "This installs nginx."
+	}
+]`)
+	})
+
+	go func() {
+		if err := http.ListenAndServe(":8080", nil); err != nil {
```
[the remark this answers is not in the document]
Comment (aclevername) [reply tests/integration/integration_suite_test.go]: done

**Review by Callisto13 — APPROVED** [review]
🎉  nice simple first imlpementation

**Pull request merged**